dataframe/series.go

```diff
@@ -0,0 +1,423 @@
+package dataframe
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.starlark.net/starlark"
+)
+
+// Series represents a sequence of values, either ints, floats, or objects. This is
+// the underlying data structure that is used to create DataFrames. A single column
+// of a DataFrame is a Series.
+type Series struct {
+	frozen    bool
+	which     int
+	valInts   []int
+	valFloats []float64
+	valObjs   []string
+	index     []string
+	dtype     string
+	name      string
+}
+
+// A Series contains values of one of these three types. The which field uses these
+// constants to determine which slice holds the actual values of the Series.
+const (
+	TypeInt   = 1
+	TypeFloat = 2
+	TypeObj   = 3
+)
+
+// Freeze prevents the series from being mutated
+func (s *Series) Freeze() {
+	s.frozen = true
+}
+
+// Hash cannot be used with Series
+func (s *Series) Hash() (uint32, error) {
+	return 0, fmt.Errorf("'Series' objects are mutable, thus they cannot be hashed")
+}
+
+// String returns the Series as a string in a readable, tabular form
+func (s *Series) String() string {
+	return s.stringify()
+}
+
+// Truth converts the series into a bool
+func (s *Series) Truth() starlark.Bool {
+	// NOTE: In python, calling bool(Series) raises this exception: "ValueError: The truth
+	// value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all()."
+	// Since starlark does not have exceptions, just always return true.
+	return true
+}
+
+// Type returns the type as a string
+func (s *Series) Type() string {
+	return "dataframe.Series"
+}
+
+// Attr gets a value for a string attribute
+func (s *Series) Attr(name string) (starlark.Value, error) {
+	return nil, starlark.NoSuchAttrError(name)
```

Comment:
You can just return `nil, nil` & starlark will create the `NoSuchAttrError` for you. That said, we need to properly implement the `HasAttrs` interface.

Attrs should return both method & attribute names, and `get` is listed as a [pandas Series method](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.get.html). All current tests are accessing via indexing, but it should also be possible to write `data frame.Series([1]).get(0)`

This test should pass, and some version of this test should be added to the suite:

```go
func TestSeriesAttrs(t *testing.T) {
	output := "\n"
	printCollect := func(thraed *starlark.Thread, msg string) {
		output = fmt.Sprintf("%s%s\n", output, msg)
	}
	thread := &starlark.Thread{
		Load:  testdata.NewModuleLoader(Module),
		Print: printCollect,
	}
	starlarktest.SetReporter(thread, t)

	prog := `
load("dataframe.star", "dataframe")

print(dir(dataframe.Series([1,2,3])))
`

	if _, err := starlark.ExecFile(thread, "", prog, nil); err != nil {
		t.Fatal(err)
	}

	expect := strings.Trim("[\"get\"]", "\n")
	output = strings.Trim(output, "\n")
	if diff := cmp.Diff(expect, output); diff != "" {
		t.Errorf("mismatch. (-want +got):\n%s", diff)
	}
}
```

starlark's string is a great reference for implementing the `HasAttrs` interface: https://github.com/google/starlark-go/blob/1cdb82c9e17a3e18b5067713955f174b08776f8b/starlark/value.go#L546-L547

Reply:
I was going to argue for implementing attributes in another PR, but you're right that `get` makes it somewhat necessary now. Added just as you described.